doc/dependencies.md

```diff
@@ -0,0 +1,36 @@
+Dependencies
+============
+
+| Dependency | Version used | Minimum Required | Latest available | CVEs? | Shared | [Bundled Qt Library](https://doc.qt.io/qt-5/configure-options.html) |
```

Comment:
Would be useful to have somewhat of an introduction here - what is this document, what should it be used for, etc.
Also that instructions for *installing* dependencies are in the appropriate `build-*.md`.